builder/swig/go.bzl

```diff
@@ -0,0 +1,201 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+def _swig_go_wrapper_impl(ctx):
+    module_name = ctx.attr.name
+
+    # Go outputs to a directory.
+    wrap_go_dir = ctx.actions.declare_directory("{}".format(module_name))
+
+    args = ctx.attr.extra_args + [
+        "-go",
+        "-outdir", wrap_go_dir.path,
+        "-package", "typedb_driver",
+        ctx.file.interface.path,
+    ]
+
+    if ctx.attr.enable_cxx:
+        wrap_src = ctx.actions.declare_file("{}_wrap.cxx".format(module_name))
+        directors_header = ctx.actions.declare_file("{}_wrap.h".format(module_name))
+        args = ["-c++", "-o", wrap_src.path, "-oh", directors_header.path] + args
+        swig_headers = [directors_header]
+    else:
+        wrap_src = ctx.actions.declare_file("{}_wrap.c".format(module_name))
+        args = ["-o", wrap_src.path] + args
+        swig_headers = []
+
+    for h in ctx.attr.lib[CcInfo].compilation_context.headers.to_list():
+        args = ["-I" + h.dirname] + args
+
+    ctx.actions.run(
+        inputs = depset(
+            [ctx.file.interface] + ctx.files.includes,
+
```

Comment:
Let's try avoid such levitation, it confuses a bit when you have a skim through the code.